malduck/compression/components/aplib.py

```diff
@@ -1,132 +1,172 @@
-        # load next tag
-        ud.tag = ord(ud.source.read(1))
-        ud.bitcount = 7
-
-    # shift bit out of tag
-    bit = ud.tag >> 7 & 0x01
-    ud.tag <<= 1
-
-    return bit
-
-
-def ap_getgamma(ud: APDSTATE) -> int:
-    result = 1
-
-    # input gamma2-encoded bits
-    while True:
-        result = (result << 1) + ap_getbit(ud)
-        if not ap_getbit(ud):
-            break
-
-    return result
-
-
-def ap_depack(source: bytes) -> bytearray:
-    ud = APDSTATE(source)
+                            for _ in range(length):
+                                self.destination.append(self.destination[-offs])
+                        else:
+                            if lwm == 0:
+                                offs -= 3
+                            else:
+                                offs -= 2
 
-    r0 = -1
-    lwm = 0
-    done = False
+                            offs <<= 8
+                            offs += ord(self.source.read(1))
+                            length = self.getgamma()
 
-    # first byte verbatim
-    ud.destination += ud.source.read(1)
+                            if offs >= 32000:
+                                length += 1
+                            if offs >= 1280:
+                                length += 1
+                            if offs < 128:
+                                length += 2
 
-    # main decompression loop
-    while not done:
-        if ap_getbit(ud):
-            if ap_getbit(ud):
-                if ap_getbit(ud):
-                    offs = 0
+                            for _ in range(length):
+                                self.destination.append(self.destination[-offs])
 
-                    for _ in range(4):
-                        offs = (offs << 1) + ap_getbit(ud)
+                            r0 = offs
 
-                    if offs:
-                        ud.destination.append(ud.destination[-offs])
-                    else:
-                        ud.destination.append(0x00)
-
-                    lwm = 0
+                        lwm = 1
                 else:
-                    offs = ord(ud.source.read(1))
+                    self.destination += self.source.read(1)
+                    lwm = 0
 
-                    length = 2 + (offs & 0x0001)
+        except (TypeError, IndexError):
+            if self.strict:
+                raise RuntimeError('aPLib decompression error')
 
-                    offs >>= 1
+        return bytes(self.destination)
 
-                    if offs:
-                        for _ in range(length):
-                            ud.destination.append(ud.destination[-offs])
-                    else:
-                        done = True
+    def pack(self):
+        raise NotImplementedError
 
-                    r0 = offs
-                    lwm = 1
-            else:
-                offs = ap_getgamma(ud)
 
-                if lwm == 0 and offs == 2:
-                    offs = r0
+def decompress(data, strict=False):
```

Comment:
I think this part (+ main) is duplicated with `compression/aplib.py` and we don't need it anymore here?

Reply:
Deleted the decompress function from aplib.py